src/coreclr/clrfeatures.cmake

```diff
@@ -38,4 +38,21 @@ endif()
 
 if (CLR_CMAKE_TARGET_WIN32)
   set(FEATURE_TYPEEQUIVALENCE 1)
-endif(CLR_CMAKE_TARGET_WIN32)
\ No newline at end of file
+endif(CLR_CMAKE_TARGET_WIN32)
+
+# Enable cached interface dispatch so that we can test/debug it more easily on non-embedded scenarios (set DOTNET_UseCachedInterfaceDispatch=1)
+# Only enable in chk/debug builds as this support isn't intended for retail use elsewhere
+if (CLR_CMAKE_TARGET_ARCH_AMD64 OR CLR_CMAKE_TARGET_ARCH_ARM64)
+  set(FEATURE_CORECLR_CACHED_INTERFACE_DISPATCH $<IF:$<CONFIG:Debug,Checked>,1,0>)
+endif()
+
+if (CLR_CMAKE_TARGET_MACCATALYST OR CLR_CMAKE_TARGET_IOS OR CLR_CMAKE_TARGET_TVOS)
+  set(FEATURE_CORECLR_CACHED_INTERFACE_DISPATCH 1)
+else()
+  set(FEATURE_CORECLR_VIRTUAL_STUB_DISPATCH 1)
+endif()
+
+if (CLR_CMAKE_HOST_UNIX AND CLR_CMAKE_HOST_ARCH_AMD64)
+  # Allow 16 byte compare-exchange (cmpxchg16b)
+  add_compile_options($<${FEATURE_CORECLR_CACHED_INTERFACE_DISPATCH}:-mcx16>)
+endif()
```

Comment:
Is this check for both UNIX/AMD64 _and_ if `FEATURE_CORECLR_CACHED_INTERFACE_DISPATCH` is set?

I'm definitely not a CMake expert, but I think this syntax means if `FEATURE_CORECLR_CACHED_INTERFACE_DISPATCH` is non-zero, then apply the flag. If so, can't we remove the explicit `if`?

My push back here if we want to expand this, then at present we need to update two locations instead of one, where `FEATURE_CORECLR_CACHED_INTERFACE_DISPATCH` is set.

Reply:
You can combine all the options into the generator expression, but it usually becomes pretty unreadable with all the $< that get accumulated. So I usually use it just for the build configs or stuff where the expression is simple.

Reply:
Ah, that's right, I've missed this point. We don't need the `if` around that.

Reply:
We don't want to set this flag on Windows or when targeting any other processor type than X64. So we still need the if. FEATURE_CORECLR_CACHED_INTERFACE_DISPATCH is sometimes enabled for Windows and it is really intended to be used for Arm64 in most cases.

Reply:
So you're saying that `FEATURE_CORECLR_CACHED_INTERFACE_DISPATCH` is valid without the 16 byte compare exchange?

Reply:
No, I'm saying that the -mcx16 switch is only valid/useful for the targeting X64 on unix platforms.  can't find ANY documentation that hints that this switch is useful on architectures other than x86 or X64. On Arm64, the baseline hardware we support requires 16 byte compare and swap, and the needed instructions are unconditionally available (via libatomic, they will either be the `ldaxp` and `stxp` pair or the `caspal` instruction, and on Mac/iOS the compiler will always general the `caspal` instruction. On Winodws, the -mcx16 switch generates a compiler warning and isn't useful, as all hardware which runs supported versions of Windows has the compare exchange 16 byte instruction.